server/src/main/java/org/elasticsearch/common/compress/CompressedXContent.java

```diff
@@ -43,43 +43,43 @@
     private static final ThreadLocal<InflaterAndBuffer> inflater1 = ThreadLocal.withInitial(InflaterAndBuffer::new);
     private static final ThreadLocal<InflaterAndBuffer> inflater2 = ThreadLocal.withInitial(InflaterAndBuffer::new);
```

Comment:
@martijnvg I just realized, we can probably remove all of this infrastructure here now that we have efficient `equals`? I added this only to speed up comparison a while back, but now it seems irrelevant and can be simplified?

Reply:
Yes, I've removed inflater2 and equalsWhenUncompressed() method: c05791c781e3ca27a56a190bb265a49770da8db3

`inflater1 ` and a few other things are still in use by production code.